src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -644,17 +644,23 @@ else if (compare > 0)
                 if (size == 0)
                     return Optional.empty();
 
-                // TODO: Find a faster implementation using binarySearch
                 l.sort(comparator);
-                for (int i = 0; i < size; i++)
-                    if (comparator.compare(value, l.get(i)) <= 0)
-                        return Optional.of((long) i);
-
-                return Optional.of((long) size);
+                int index = binarySearchRank(l, value, comparator);
+                return Optional.of((long) index);
             }
         );
     }
 
+    private static <U> int binarySearchRank(List<? extends U> l, U value, Comparator<? super U> comparator) {

```

Comment:
I created this method and chose name `binarySearchRank` because:
1) it requires list to be sorted just like `Collections.binarySearch`,
2) it does not return negative indices like original, but rather returns positive rank.